bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/handler/capability/RefreshCapability.java

```diff
@@ -12,114 +12,75 @@
+        }
     }
 
     @Override
     public void initialize() {
-        this.refreshConfigured = !probing();
-        freeJobAndReschedule(2);
+        expireData();
     }
 
     @Override
     public void dispose() {
-        freeJobAndReschedule(0);
+        freeJobAndReschedule(null);
         super.dispose();
     }
 
     @Override
     public void expireData() {
-        dataTimeStamp = Instant.now().minus(dataValidity);
-        freeJobAndReschedule(1);
-    }
-
-    private Duration dataAge() {
-        return Duration.between(dataTimeStamp, Instant.now());
-    }
-
-    private boolean probing() {
-        return dataValidity.getSeconds() <= 0;
+        freeJobAndReschedule(ASAP);
     }
 
     private void proceedWithUpdate() {
         handler.proceedWithUpdate();
-        long delay;
+        Duration delay = calcDelay();
         if (!ThingStatus.ONLINE.equals(handler.getThing().getStatus())) {
-            logger.debug("{} is not ONLINE, special refresh interval is used", thingUID);
-            delay = OFFLINE_INTERVAL.toSeconds();
-            if (probing()) {
-                dataTimeStamp0 = Instant.MIN;
-            }
-        } else {
-            delay = refreshConfigured ? dataValidity.getSeconds()
-                    : (probing() ? PROBING_INTERVAL : dataValidity.minus(dataAge()).plus(DEFAULT_DELAY)).toSeconds();
+            logger.debug("Thing '{}' is not ONLINE, using special refresh interval", thingUID);
+            delay = OFFLINE_DELAY;
         }
-        delay = delay < 2 ? PROBING_INTERVAL.toSeconds() : delay;
-        logger.debug("{} refreshed, next one in {}s", thingUID, delay);
+        logger.debug("'{}' refreshed, next one in {}", thingUID, delay);
         freeJobAndReschedule(delay);
     }
 
-    @Override
-    protected void updateNAThing(NAThing newData) {
-        super.updateNAThing(newData);
-        newData.getLastSeen().map(ZonedDateTime::toInstant).ifPresent(tsInstant -> {
-            if (probing()) {
-                if (Instant.MIN.equals(dataTimeStamp0)) {
-                    dataTimeStamp0 = tsInstant;
-                    logger.debug("First data timestamp of {} is {}", thingUID, dataTimeStamp0);
-                } else if (tsInstant.isAfter(dataTimeStamp0)) {
-                    dataValidity = Duration.between(dataTimeStamp0, tsInstant);
-                    refreshConfigured = true;
-                    logger.debug("Data validity period of {} identified to be {}", thingUID, dataValidity);
-                } else {
-                    logger.debug("Data validity period of {} not yet found, data timestamp unchanged", thingUID);
-                }
-            }
-            dataTimeStamp = tsInstant;
-        });
+    protected Duration calcDelay() {
+        return dataValidity;
     }
 
-    private void freeJobAndReschedule(long delay) {
+    private void freeJobAndReschedule(@Nullable Duration delay) {
```

Comment:
This method has been changed a few times, so I've also looked at it quite a few times. Even though it's a small method, it actually does two different things (as the method name also implies):
- When `delay` is null, it interrupts the currently running job.
- When `delay` is not null, it reschedules the job.

We could consider splitting it into two distinct methods for better readability (especially reading the calling methods would be more self-explanatory), i.e. something like:
```java
private void rescheduleJob(Duration delay) {
    refreshJob.ifPresent(job -> job.cancel(true));
    refreshJob = handler.schedule(() -> proceedWithUpdate(), delay);
}

private void stopJob() {
    refreshJob.ifPresent(job -> job.cancel(true));
    refreshJob = Optional.empty();
}
```

Reply:
Done